### DISCUSSION TITLE
Pressure levels / Geopotential heights forecast implementation

### DISCUSSION THREAD

**Discussion by MantraMedia** [open-meteo/open-meteo#588] [General]
I see that cmc_gem_gdps for example has 180 different variables , most of them geopotential heights / pressure levels.

Are these necessary values for the calculation of the wind, temperature, weather forecast etc or can they be omitted?

**Comment by patrick-zippenfenig** [comment]
They can be omitted. Pressure level data is used to calculate variables like `cloud_cover_low/mid/high` if the weather model does not offer them directly (e.g. open data ECMWF IFS), but is stored afterwards in separate files. In the future, more derived variable like freezing level or snowfall height could be calculated for more models.

Just for reference for other: Here is the overview of `cmc_gem_gdps` in question: https://openmeteo.s3.amazonaws.com/index.html#data/cmc_gem_gdps/

**Comment by MantraMedia** [comment]
thank you for the fast reply , if I understand it correctly , this also means that for https://openmeteo.s3.amazonaws.com/index.html#data/ncep_gfs025/ all variables are needed to get  cloud cover, temperature_2m , rh_2m, wind_speed_10, wind_gust_10_m reliably as these values are not available?

**Reply by patrick-zippenfenig** [reply]
That's a special case ;-). The API combines `GFS025` and `GFS013` for any API call using `GFS Global`. Because wind speeds at 10 are available for `GFS013`, I do not process them for `GFS025`.
